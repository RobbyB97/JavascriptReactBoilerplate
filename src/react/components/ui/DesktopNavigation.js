/* Packages */
import React from 'react';
import {NavLink} from 'react-router-dom';
import {connect} from 'react-redux';


/* Component */
export class DesktopNavigation extends React.Component {
    constructor(props) {
        super(props);
    };

    render() {
        return (
            <section id="DesktopNavigation">
                <ul className="DesktopNavigation__list">
                    {this.props.internal_links &&
                        this.props.internal_links.map((link) => (
                            <li key={link.text}>
                                <a href={link.href}>
                                    {link.text}
                                </a>
                            </li>
                        ))
                    }
                </ul>
            </section>
        );
    };
};


export default DesktopNavigation;