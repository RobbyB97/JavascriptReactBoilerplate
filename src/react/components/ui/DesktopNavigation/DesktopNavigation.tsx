import React from 'react';

import DesktopNavigationLink from './DesktopNavigationLink';
import {navigationLinksType} from '../../../data/navigationLinks'


interface Props {
    links: navigationLinksType[];
}

interface State {

}


export class DesktopNavigation extends React.Component<Props, State> {
    constructor(props: Props) {
        super(props);
    };

    render() {
        return (
            <section id="DesktopNavigation">
                <nav className="DesktopNavigation__nav">
                    <ul className="DesktopNavigation__list">
                        {/* Links */}
                        {this.props.links &&
                            this.props.links.map((link) => (
                                <DesktopNavigationLink 
                                    key={link.text}
                                    {...link}
                                />
                            ))
                        }                 
                    </ul>
                </nav>
            </section>
        );
    };
};


export default DesktopNavigation;