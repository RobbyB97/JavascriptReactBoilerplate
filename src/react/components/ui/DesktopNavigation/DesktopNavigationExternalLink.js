/* Packages */
import React from 'react';
import {connect} from 'react-redux';


/* Component */
export class DesktopNavigationExternalLink extends React.Component {
    constructor(props) {
        super(props);
    };


    render() {
        return (
            <li className="DesktopNavigationLink">
                <a
                    className="DesktopNavigationLink__link"
                    href={this.props.href}
                    target="_blank"
                    rel="noopener"
                >
                    {this.props.text}
                </a>

                {/* Sublink menu */}
                {this.props.sublinks &&
                    <ul className="DesktopNavigationLink__sublink_list">
                        {this.props.sublinks.map((sublink) => (
                            <li
                                className="DesktopNavigation__sublink_link"
                                key={sublink.text}
                            >
                                <a href={sublink.href}>
                                    {sublink.text}
                                </a>
                            </li>
                        ))

                        }
                    </ul>
                }
            </li>
        );
    };
};


export default DesktopNavigationExternalLink;