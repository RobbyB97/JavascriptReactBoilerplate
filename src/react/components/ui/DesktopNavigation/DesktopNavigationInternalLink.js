/* Packages */
import React from 'react';
import {NavLink} from 'react-router-dom';


/* Component */
export class DesktopNavigationInternalLink extends React.Component {
    constructor(props) {
        super(props);
    };

    render() {
        return (
            <li className="DesktopNavigationLink">
                <NavLink
                    className="DesktopNavigationLink__link"
                    to={this.props.route}
                >
                    {this.props.text}
                </NavLink>

                {/* Sublink menu */}
                {this.props.sublinks &&
                    <ul
                        className="DesktopNavigationLink__sublink_list"
                    >
                        {this.props.sublinks.map((sublink) => (
                            <li
                                className="DesktopNavigationLink__sublink_link"
                                key={sublink.text}
                            >
                                <NavLink to={sublink.route}>
                                    {sublink.text}
                                </NavLink>
                            </li>
                        ))}
                    </ul>
                }
            </li>
        );
    };
};


export default DesktopNavigationInternalLink;