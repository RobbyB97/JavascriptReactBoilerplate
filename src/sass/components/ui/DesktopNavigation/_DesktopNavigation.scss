#DesktopNavigation {
    display: none;

    @media #{$desktop_breakpoint} {
        display: block;
    }
}


.DesktopNavigation__nav {
    position: fixed;
    right: 1.5em;
    top: 2em;
    z-index: 8;
}


.DesktopNavigation__list {
    padding: 0;
    list-style-type: none;
    display: flex;
    flex-direction: row;
    margin-top: 0;
    margin-bottom: 0;
    width: calc(100vw - 120px);
    flex-wrap: wrap;
    justify-content: flex-end;
}


.DesktopNavigation__link {
    a {
        color: $white;
        transition: 100ms;
        text-decoration: none;
        opacity: 0.9;
        font-size: $s_font;
        border-bottom: 0px solid $white;

        &:hover {
            transition: 100ms;
            opacity: 1;
            border-bottom: 2px solid $white;
        }
    }

    &:not(:last-child) {
        margin-right: $m-font;
    }
}