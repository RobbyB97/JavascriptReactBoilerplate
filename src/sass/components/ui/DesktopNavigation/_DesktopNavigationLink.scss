.DesktopNavigationLink {
    transition: 100ms;
    min-width: 120px;

    &:not(:last-child) {
        margin-right: 2em;
    }
}


/* 'Active' state */
.DesktopNavigationLink:hover {
    .DesktopNavigationLink__link {
        border-bottom: 2px solid $white;
    }

    .DesktopNavigationLink__sublink_list {
        height: auto;
    }
}


.DesktopNavigationLink__link {
    color: $white;
    text-decoration: none;
    transition: 100ms;
    display: block;
    text-align: center;
}


.DesktopNavigationLink__sublink_list {
    list-style-type: none;
    overflow: hidden;
    background: $shadow;
    padding: 0;
    height: 0;
}


.DesktopNavigationLink__sublink {
    &:not(:last-child) {
        border-bottom: 1px solid $shadow;
    }
}


.DesktopNavigationLink__sublink_link {
    color: $white;
    text-decoration: none;
    display: block;
    padding: 8px;
    text-align: center;
    font-size: $s_font;
    
    &:hover {
        background: $shadow_thick;
    }
}