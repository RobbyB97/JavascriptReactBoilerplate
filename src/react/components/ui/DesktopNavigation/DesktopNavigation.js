/* Packages */
import React from 'react';


/* Components */
import DesktopNavigationInternalLink from './DesktopNavigationInternalLink';
import DesktopNavigationExternalLink from './DesktopNavigationExternalLink';


/* Component */
export class DesktopNavigation extends React.Component {
    constructor(props) {
        super(props);
    };

    render() {
        return (
            <section id="DesktopNavigation">
                <nav className="DesktopNavigation__nav">
                    <ul className="DesktopNavigation__list">
                        {/* Internal links */}
                        {this.props.internal_links &&
                            this.props.internal_links.map((link) => (
                                <DesktopNavigationInternalLink {...link}/>
                            ))
                        }

                        {/* External links */}
                        {this.props.external_links &&
                            this.props.external_links.map((link) => (
                                <DesktopNavigationExternalLink {...link}/>
                            ))
                        }                    
                    </ul>
                </nav>
            </section>
        );
    };
};


export default DesktopNavigation;