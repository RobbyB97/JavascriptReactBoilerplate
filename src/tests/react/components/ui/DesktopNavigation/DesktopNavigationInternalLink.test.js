/* Packages */
import React from 'react';
import {shallow} from 'enzyme';


/* Component */
import {DesktopNavigationInternalLink} from '../../../../../react/components/ui/DesktopNavigation/DesktopNavigationInternalLink'


/* Test config */
let component;
beforeEach(() => {
    component = shallow(<DesktopNavigationInternalLink />)
});


/* Tests */
test('Render DesktopNavigationInternalLink component', () => {
    expect(component).toMatchSnapshot();
})