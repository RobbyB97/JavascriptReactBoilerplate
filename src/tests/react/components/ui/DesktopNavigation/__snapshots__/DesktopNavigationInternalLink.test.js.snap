// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render DesktopNavigationInternalLink component 1`] = `
<li
  className="DesktopNavigationLink"
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="DesktopNavigationLink__link"
  />
</li>
`;
